components/Chips/src/MDCChipField.h

```diff
@@ -0,0 +1,242 @@
+ http://stackoverflow.com/a/25862878/23624.
+
+ To enable the workaround, define MDC_CHIPFIELD_PRIVATE_API_BUG_FIX to be 1 when building
+ MDCChipField. MDC itself will never define this macro, it's entirely a client decision.
+
+ If you target iOS versions < 8.3 and you decide not to define MDC_CHIPFIELD_PRIVATE_API_BUG_FIX,
+ your iOS 8.0-8.2 users will *not be able to delete chips using the backspace key*. Please
+ consider adding a delete button to each chip's accessoryView.
+
+ An alternative approach (using an empty character to detect backspace) makes MDCChipField unusable
+ by VoiceOver users on all versions of iOS.
+ */
+
+/** The minimum text field width of the text field contained within the MDCChipField. */
+extern const CGFloat kMDCChipFieldDefaultMinTextFieldWidth;
+
+/** The default content edge insets for the MDCChipField. */
+extern const UIEdgeInsets kMDCChipFieldDefaultContentEdgeInsets;
+
+@protocol MDCChipFieldDelegate;
+
+/** Defines when text field input is changed into a chip. */
+typedef NS_OPTIONS(NSUInteger, MDCChipFieldDelimiter) {
+  MDCChipFieldDelimiterNone = 0,
+  MDCChipFieldDelimiterReturn = 1 << 0,
+  MDCChipFieldDelimiterSpace = 1 << 1,
+  MDCChipFieldDelimiterDidEndEditing = 1 << 2,
+  MDCChipFieldDelimiterDefault = (MDCChipFieldDelimiterReturn | MDCChipFieldDelimiterDidEndEditing),
+  MDCChipFieldDelimiterAll = 0xFFFFFFFF
+};
+
+@interface MDCChipField : UIView
+
+/**
+ The text field used to enter new chips.
+
+ Do not set the delegate or positioningDelegate.
+
+ If you set a custom font, make sure to also set the custom font on textField.placeholderLabel and
+ on your MDCChipView instances.
+ */
+@property(nonatomic, nonnull, readonly) MDCTextField *textField;
+
+/**
+ The fixed height of all chip views.
+
+ Default is 32.
+ */
+@property(nonatomic, assign) CGFloat chipHeight;
+
+/**
+ Attribute to determine whether to show the placeholder text (if it exists) when chips are
+ present.
+
+ Default is YES.
+ */
+@property(nonatomic, assign) BOOL showPlaceholderWithChips;
+
+/**
+ The delimiter used to create chips in the text field. Uses default value
+ MDCChipFieldDelimiterDefault if no delimiter is set.
+ */
+@property(nonatomic, assign) MDCChipFieldDelimiter delimiter;
+
+/**
+ The minimum width of the text field.
+
+ Default is |kMDCChipFieldDefaultMinTextFieldWidth|.
+ */
+@property(nonatomic, assign) CGFloat minTextFieldWidth;
+
+/**
+ The chips that are visible in the input area.
+ */
+@property(nonatomic, nonnull, copy) NSArray<MDCChipView *> *chips;
+
+/**
+ Delegate to receive updates to the chip field view. Implement
+ |chipFieldHeightDidChange| to receive updates when the height of the chip field changes.
+ */
+@property(nonatomic, nullable, weak) id<MDCChipFieldDelegate> delegate;
+
+/**
+ The inset or outset margins for the rectangle surrounding all of the chip field's content.
+ Default is |kMDCChipFieldDefaultContentEdgeInsets|.
```

Comment:
We seem to be prefixing k to constants in some components while omitting k in others. Use latter style?

Reply:
Removed k prefix. This was actually breaking the extern'd consts because only the header used the k prefix!